docs/en/latest/plugins/ai-rag.md

```diff
@@ -0,0 +1,202 @@
+---
+title: ai-rag
+keywords:
+  - Apache APISIX
+  - API Gateway
+  - Plugin
+  - ai-rag
+description: This document contains information about the Apache APISIX ai-rag Plugin.
+---
+
+<!--
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+-->
+
+## Description
+
+The `ai-rag` plugin integrates Retrieval-Augmented Generation (RAG) capabilities with AI models.
+It allows efficient retrieval of relevant documents or information from external data sources and
+augments the AI's responses with that data, improving the accuracy and context of generated outputs.
+
+**_This plugin must be used in routes that proxy requests to LLMs only._**
```

Comment:
Same question as the other plugin: only on `routes`? not other places that can configure plugins? 🤔